ethpm/package.py

```diff
@@ -215,5 +222,50 @@ def deployments(self) -> Union["Deployments", Dict[None, None]]:
             )
             for deployment_data in deployments.values()
         }
+        validate_deployments_tx_receipt(deployments, self.w3)
+        linked_deployments = get_linked_deployments(deployments)
+        if linked_deployments:
+            for deployment_data in linked_deployments.values():
+                on_chain_bytecode = self.w3.eth.getCode(
+                    to_checksum_address(deployment_data["address"])
+                )
+                unresolved_linked_refs = normalize_linked_references(
+                    deployment_data["runtime_bytecode"]["link_dependencies"]
+                )
+                resolved_linked_refs = tuple(
+                    self._resolve_linked_references(link_ref, deployments)
+                    for link_ref in unresolved_linked_refs
+                )
+                validate_linked_references(resolved_linked_refs[0], on_chain_bytecode)
 
         return Deployments(deployments, all_contract_factories, self.w3)
+
+    @to_tuple
+    def _resolve_linked_references(
+        self, link_ref: Tuple[int, str, str], deployments: Dict[str, Any]
+    ) -> Generator[Tuple[int, bytes], None, None]:
+        # No nested deployment: i.e. 'Owned'
+        offset, link_type, value = link_ref
+        if link_type == "literal":
+            yield offset, to_canonical_address(value)
+        elif value in deployments:
+            yield offset, to_canonical_address(deployments[value]["address"])
+        # No nested deployment, but invalid ref
+        elif ":" not in value:
+            raise BytecodeLinkingError(
+                "Contract instance reference: {0} not found in package's deployment data.".format(
+                    value
+                )
+            )
+        # Expects child pkg in build_dependencies
+        elif value.split(":")[0] not in self.build_dependencies:
+            raise BytecodeLinkingError(
+                "Expected build dependency: {0} not found in package's build dependencies.".format(
+                    value.split(":")[0]
+                )
+            )
+        # Find and return resolved, nested ref
+        else:
+            unresolved_linked_ref = value.split(":", 1)[-1]
+            build_dependency = self.build_dependencies[value.split(":")[0]]
+            return build_dependency._resolve_link_dependencies(unresolved_linked_ref)
```

Comment:
Confusing python thing.  `return` statements in generators don't return anything.

```python
In [7]: def a_generator(v):
   ...:     if v:
   ...:         yield "v is true"
   ...:     else:
   ...:         return "v is false"
   ...:

In [9]: tuple(a_generator(True))
Out[9]: ('v is true',)

In [10]: tuple(a_generator(False))
Out[10]: ()
```

These either all need to be `yield` statements or `return` statements and it looks like they could all be changed to be `return` statements.